pyproject.toml

```diff
@@ -1,3 +1,58 @@
+[build-system]
+requires = ["setuptools>=62.3.0", "wheel"]
```

Comment:
let's be careful with the setuptools version... was it already pinned like this? 
latest setuptools versions changed some things

some context:

> setuptools in turn has recently switched to adhering to PEP 625, which enforces that hyphens in package names be normalized to underscores to eliminate a parsing ambiguity in the sdist file format.

this impacts the dot that it's in the package name 

https://peps.python.org/pep-0625/
https://packaging.python.org/en/latest/specifications/binary-distribution-format/#escaping-and-unicode
https://github.com/pypa/setuptools/issues/3593
https://github.com/pypa/twine/pull/1096

Reply:
Interesting. I wasn't aware of the new filename normalization. Thanks for the heads up. I don't think setuptools was pinned at all, but if I look in the last deploy that happened, I see `setuptools` `65.5.0` being installed.

I imagine going to a newer `setuptools` might need some adaptation in the script we use on the feedstock `https://github.com/conda-forge/lumicks.pylake-feedstock/blob/main/recipe/meta.yaml`.

What would you suggest? Upper bound it to `69.3`for now or cut the cord?

It should only impact generated file names though, right?

Reply:
within this PR, I'd suggest pinning an upper bound, and then investigate the potential impact of upgrading setuptools in a separate PR

Reply:
We noticed this behavior in other projects, where we have CI workflows for testing the packaging and what goes inside the distribution, so might be interested to look into that as well